### PR TITLE
i18n(zh-cn): Translate `recipes/i18n.mdx` and update `core-concepts/routing.mdx`

### PR DESCRIPTION
**Pull request by 100gle**
#### What kind of changes does this PR include?

- Translated content
- Update existed content

#### Description

- Translate `recipes/i18n.mdx` content to zh-CN 
- Update `core-concepts/routing.mdx` with `redirects`

### CODE REVIEW

**Comment by 100gle** [comment]
cc @liruifengv

File: src/content/docs/zh-cn/recipes/i18n.mdx [hunk 1]
```diff
@@ -0,0 +1,332 @@
+---
+title: 添加 i18n 功能
+description: 通过动态路由和文档集来让你的 Astro 站点支持国际化。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/recipes/i18n.mdx:3]:
```suggestion
description: 通过动态路由和文档集来让你的 Astro 站点支持国际化。
```
```suggestion
description: 通过动态路由和内容集合来让你的 Astro 站点支持国际化。
```

File: src/content/docs/zh-cn/recipes/i18n.mdx [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+type: recipe
+i18nReady: true
+---
+import FileTree from '~/components/FileTree.astro'
+import StaticSsrTabs from '~/components/tabs/StaticSsrTabs.astro'
+
+尽管 Astro 没有内置的国际化功能，但你也可以构建你自己的国际化解决方案。
+
+本节示例中，你将看到如何使用文档集和动态路由来支持不同语言。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/recipes/i18n.mdx:12]:
```suggestion
本节示例中，你将看到如何使用内容集合和动态路由来支持不同语言。
```

File: src/content/docs/zh-cn/recipes/i18n.mdx [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+        // src/pages/index.astro
+        ---
+        <meta http-equiv="refresh" content="0;url=/en/" />
+        ```
+
+        这种方式通过 [meta refresh](https://en.wikipedia.org/wiki/Meta_refresh) 来实现且不会受限于你的部署方式；不过有些静态服务器也允许你通过自定义配置文件来让你的服务器重定向，这可能需要你进一步参考你所使用的部署平台文档。
+      </Fragment>
+      
+      <Fragment slot="ssr">
+        如果你正好使用 SSR 适配器，那么你也可以使用 [`Astro.redirect`](/zh-cn/core-concepts/routing/#dynamic-redirects) 来重定向到默认语言。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/recipes/i18n.mdx]: There will be a broken link.

File: src/content/docs/zh-cn/recipes/i18n.mdx [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+        ```astro
+        ---
+        // src/pages/index.astro
+        return Astro.redirect('/en/');
+        ---
+        ```
+      </Fragment>
+    </StaticSsrTabs>
+
+### 为翻译内容创建文档集
```
Comment (liruifengv) [inline src/content/docs/zh-cn/recipes/i18n.mdx:60]:
```suggestion
### 为翻译内容创建内容集合
```

File: src/content/docs/zh-cn/recipes/i18n.mdx [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+
+1. 在 `src/content` 文件夹中为每种你想要包含的内容创建一个文件夹，同时在其中也创建对应语言的子目录。假设你现在打算为博客文章支持英语和法语：
+
+    <FileTree>
+    - src/
+      - content/
+          - blog/
+            - **en/** Blog posts in English
```
Comment (liruifengv) [inline src/content/docs/zh-cn/recipes/i18n.mdx:68]: There also need to translate.

File: src/content/docs/zh-cn/recipes/i18n.mdx [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+                - post-1.md
+                - post-2.md
+            - **fr/** Blog posts in French
```
Comment (liruifengv) [inline src/content/docs/zh-cn/recipes/i18n.mdx:71]: There is also need to translate.

File: src/content/docs/zh-cn/recipes/i18n.mdx [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+              - post-1.md
+              - post-2.md
+    </FileTree>
+
+2. 创建一个 `src/content/config.ts` 文件并且导出对应的文档集。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/recipes/i18n.mdx:76]:
```suggestion
2. 创建一个 `src/content/config.ts` 文件并且导出对应的内容集合。
```

File: src/content/docs/zh-cn/recipes/i18n.mdx [hunk 8]
```diff
@@ -0,0 +1,332 @@
...
+        date: z.date()
+      })
+    });
+
+    export const collections = {
+      'blog': blogCollection
+    };
+
+    ```
+    📚 你可以在这获取有关更多 [文档集](/zh-cn/guides/content-collections/) 的内容。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/recipes/i18n.mdx:95]:
```suggestion
    📚 你可以在这获取有关更多 [内容集合](/zh-cn/guides/content-collections/) 的内容。
```

File: src/content/docs/zh-cn/recipes/i18n.mdx [hunk 9]
```diff
@@ -0,0 +1,332 @@
...
+        <Content/>
+        ```
+      </Fragment>
+    </StaticSsrTabs>
+
+    📚 你可以在这获取有关更多 [动态路由](/zh-cn/core-concepts/routing/#动态路由) 的内容。
+
+
+    :::tip[日期格式化]
+    在上述例子中使用了内置的 [`toLocaleString()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString) 日期格式化方法来创建为人类可读的日期字符串，以确保日期和时间格式与用户所选择的语言相匹配。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/recipes/i18n.mdx:160]:
```suggestion
    在上述例子中使用了内置的 [`toLocaleString()`](https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString) 日期格式化方法来创建为人类可读的日期字符串，以确保日期和时间格式与用户所选择的语言相匹配。
```

MDN link can use zh-CN

File: src/content/docs/zh-cn/recipes/i18n.mdx [hunk 10]
```diff
@@ -0,0 +1,332 @@
...
+    :::
+
+### 翻译 UI 标签
+
+创建条目字典来翻译你网站上用户界面的元素标签，这样可以让访问者在他们的语言环境下更好地体验你的网站。
```
Comment (liruifengv) [inline src/content/docs/zh-cn/recipes/i18n.mdx:165]:
```suggestion
创建术语字典来翻译你网站上用户界面的元素标签，这样可以让访问者在他们的语言环境下更好地体验你的网站。
```

**Review by liruifengv — CHANGES_REQUESTED** [review]

File: src/content/docs/zh-cn/recipes/i18n.mdx (hunk 3 above)
Comment (100gle) [reply src/content/docs/zh-cn/recipes/i18n.mdx]: @liruifengv Should I update this broken link content in this PR or use a new PR to update that?

**Comment by liruifengv** [comment]
`routing.mdx` 页面需要完整更新

**Review by liruifengv — APPROVED** [review]
LGTM!

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Thanks peeps 🚀

**Pull request merged**